state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1901,34 +1958,15 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 
 		let channel_owner = deposit_channel_details.owner.clone();
 
-		if TaintedTransactions::<T, I>::take(channel_owner.clone(), deposit_details.clone())
-			.is_some()
-		{
-			let refund_address = match deposit_channel_details.action {
-				ChannelAction::Swap { refund_params, .. } =>
-					refund_params.as_ref().map(|refund_params| refund_params.refund_address.clone()),
-				ChannelAction::CcmTransfer { refund_params, .. } =>
-					refund_params.as_ref().map(|refund_params| refund_params.refund_address.clone()),
-				ChannelAction::LiquidityProvision { refund_address, .. } => Some(refund_address),
-			};
-
-			let tainted_transaction_details = TaintedTransactionDetails {
-				refund_address,
-				amount: deposit_amount,
-				asset,
-				tx_id: deposit_details.clone(),
-			};
-
-			ScheduledTxForReject::<T, I>::append(tainted_transaction_details);
-
-			Self::deposit_event(Event::<T, I>::DepositIgnored {
-				deposit_address: deposit_address.clone(),
+		if deposit_channel_details.boost_status == BoostStatus::NotBoosted &&
```

Comment:
Would it be useful to log a warning if a deposit is tainted, but we had to accept it since we already boosted it?